### PR TITLE
Add winston as middleware for express

### PR DESCRIPTION
**Pull request by solker**
#19

I still don't understand some concepts.
Are we should use two logs module? One as middlewire for express, second as simple logger for everythink else. Or we can create one logger for everythink?
For me it's looking like abstract logger layer with implementation of two interfaces.  
 
Now all requests displaying in console and sending to loggly.
Also don't understand how to work next line `app.use(logging.errorLogger);`

### CODE REVIEW

**Comment by solker** [comment]
I change `morgan` logger in middleware of express to custom `logger` and save it in `lib/logger.js`. It's implement all console and loggly transport. Work as simple log and middleware for express. 

I saved little debug logger to debug main functionality.

To run in debug mode need run app with DEBUG=* environment.  Or with only specific prefix like DEBUG=app. For more documentation need read [`debug`](https://www.npmjs.com/package/debug) module. 

Because `*` also true then it is also work for winston logger.

selection_114

Branch already merged with #32 (Passing UUID to loggly) and #34 (Config)

File: config.json.example [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "DEBUG": false,
+
+  "BACKEND_URL": "http://192.168.1.32/shoov.local/",
+
+  "LOGGLY_TOKEN": "704620e7-c3d4-4d84-a0c1-926e5147592c",
```
Comment (amitaibu) [inline config.json.example]: this shouldn't hold any Gizra related info. just replace with `<YOUR-LOGGLY-TOKEN>`, `<YOUR-LOGGLY-SUBDOMAIN>`, etc'

**Comment by amitaibu** [comment]
@solker needs a re-roll

**Comment by solker** [comment]
@amitaibu done.

**Comment by amitaibu** [comment]
tests failing

**Comment by solker** [comment]
@amitaibu it's bug.

File: lib/logger.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    });
+
+    winstonTransports.transports.push(logglyTransport);
+  }
+
+  // Init a logger.
+  var log = new(winston.Logger)(winstonTransports);
+  log.addRewriter(function(level, msg, meta) {
+    // Do not modify debug level.
+    if (level === 'debug') return;
```
Comment (amitaibu) [inline lib/logger.js]: remove - lets always add a uuid

File: routes/ci.js [hunk 3]
```diff
@@ -349,6 +334,12 @@ var execDocker = function(buildId, buildItemId, accessToken, withSelenium) {
         'name': CIBuildContainerName
       };
 
+      if (debug) {
```
Comment (amitaibu) [inline routes/ci.js]: still needed?

File: lib/logger.js (hunk 2 above)
Comment (solker) [inline lib/logger.js]: debug level doesn't upload to loggly

**Comment by amitaibu** [comment]
Looking good, I've added a few minor comments.

Worth also adding a line or two in the README about `cp config.json.example config.json`

File: lib/logger.js (hunk 2 above)
Comment (amitaibu) [inline lib/logger.js]: ok, so just please change to 

```js
if (level === 'debug') {
  // Debug level isn't uploaded to Loggly.
  return;
}

File: routes/ci.js (hunk 3 above)
Comment (solker) [inline routes/ci.js]: yes, good for debug.

**Comment by solker** [comment]
@amitaibu I updated documentation and fix bugs.

**Comment by amitaibu** [comment]
Merged, thanks.

**Pull request closed without merging**